File: src/main/java/com/DavidM1A2/afraidofthedark/client/particle/ParticleSpellHit.java
package com.DavidM1A2.afraidofthedark.client.particle;

import com.DavidM1A2.afraidofthedark.common.constants.ModSprites;
import net.minecraft.world.World;

/**
 * A special particle spell hit effect
 */
public class ParticleSpellHit extends AOTDParticle
{
    /**
     * Constructor takes the x,y,z position of the particle and the world
     *
     * @param worldIn  The world the particle is at
     * @param xCoordIn The x position of the basic attack
     * @param yCoordIn The y position of the basic attack
     * @param zCoordIn The z position of the basic attack
     */
    public ParticleSpellHit(World worldIn, double xCoordIn, double yCoordIn, double zCoordIn)
    {
        super(worldIn, ModSprites.SPELL_HIT, xCoordIn, yCoordIn, zCoordIn, 0, 0, 0);
        // 0.5-1.5 second lifespan
        this.particleMaxAge = this.rand.nextInt(30) + 10;
        // Make the particles noticable
        this.particleScale = 1.0f + this.rand.nextFloat() * 4;
        // Random motion
        this.motionX = (this.rand.nextFloat() - 0.5) * 0.05;
        this.motionY = this.rand.nextFloat() * 0.02;
        this.motionZ = (this.rand.nextFloat() - 0.5) * 0.05;
    }

    /**
     * Called before the particle is moved, update the motionXYZ here
     */
    @Override
    void updateMotionXYZ()
    {
        // Don't update motion
    }
}

File: src/main/java/com/davidm1a2/afraidofthedark/client/particle/ParticleSpellHit.kt
package com.davidm1a2.afraidofthedark.client.particle

import com.davidm1a2.afraidofthedark.common.constants.ModSprites
import net.minecraft.world.World

/**
 * A special particle spell hit effect
 *
 * @constructor takes the x,y,z position of the particle and the world
 * @param world  The world the particle is at
 * @param x The x position of the spell hit effect
 * @param y The y position of the spell hit effect
 * @param z The z position of the spell hit effect
 */
class ParticleSpellHit(
    world: World,
    x: Double,
    y: Double,
    z: Double
) : AOTDParticle(world, ModSprites.SPELL_HIT, x, y, z)
{
    init
    {
        // 0.5-1.5 second lifespan
        particleMaxAge = rand.nextInt(30) + 10
        // Make the particles noticable
        particleScale = 1.0f + rand.nextFloat() * 4
        // Random motion
        motionX = (rand.nextFloat() - 0.5) * 0.05
        motionY = rand.nextFloat() * 0.02
        motionZ = (rand.nextFloat() - 0.5) * 0.05
    }
}